src/progress.rs

```diff
@@ -0,0 +1,55 @@
+use std::io::Read;
+
+use indicatif::{ProgressBar, ProgressStyle};
+use reqwest::blocking::Response;
+
+pub struct ResponseProgress {
+    progress: Option<ProgressBar>,
+    response: Response,
+}
+
+fn make_progress_bar(size: u64) -> ProgressBar {
+    let bar = ProgressBar::new(size);
+
+    bar.set_style(
+        ProgressStyle::default_bar()
+            .template("[{elapsed_precise}] [{bar:40}] {bytes}/{total_bytes} ({bytes_per_sec}, {eta})")
+            .unwrap()
```

Comment:
This can return a `TemplateError` from indicatif when the template has a syntax issue. Unwrapping because it's a static template.